src/lib/gadgets/gadgets.ts

```diff
@@ -39,6 +52,80 @@ const Gadgets = {
   rangeCheck64(x: Field) {
     return rangeCheck64(x);
   },
+
+  /**
+   * Asserts that the input value is in the range [0, 2^32).
+   *
+   * This function proves that the provided field element can be represented with 32 bits.
+   * If the field element exceeds 32 bits, an error is thrown.
+   *
+   * @param x - The value to be range-checked.
+   *
+   * @throws Throws an error if the input value exceeds 32 bits.
+   *
+   * @example
+   * ```ts
+   * const x = Provable.witness(Field, () => Field(12345678n));
+   * Gadgets.rangeCheck32(x); // successfully proves 32-bit range
+   *
+   * const xLarge = Provable.witness(Field, () => Field(12345678901234567890123456789012345678n));
+   * Gadgets.rangeCheck32(xLarge); // throws an error since input exceeds 32 bits
+   * ```
+   *
+   * **Note**: Small "negative" field element inputs are interpreted as large integers close to the field size,
+   * and don't pass the 32-bit check. If you want to prove that a value lies in the int32 range [-2^31, 2^31),
+   * you could use `rangeCheck32(x.add(1n << 31n))`.
+   */
+  rangeCheck32(x: Field) {
+    return rangeCheck32(x);
+  },
+
+  /**
+   * Asserts that the input value is in the range [0, 2^n). `n` must be a multiple of 16.
+   *
+   * This function proves that the provided field element can be represented with `n` bits.
+   * If the field element exceeds `n` bits, an error is thrown.
+   *
+   * @param x - The value to be range-checked.
+   * @param n - The number of bits to be considered for the range check.
+   * @param message - Optional message to be displayed when the range check fails.
+   *
+   * @throws Throws an error if the input value exceeds `n` bits.
+   *
+   * @example
+   * ```ts
+   * const x = Provable.witness(Field, () => Field(12345678n));
+   * Gadgets.rangeCheckN(32, x); // successfully proves 32-bit range
+   *
+   * const xLarge = Provable.witness(Field, () => Field(12345678901234567890123456789012345678n));
+   * Gadgets.rangeCheckN(32, xLarge); // throws an error since input exceeds 32 bits
+   * ```
+   */
+  rangeCheckN(n: number, x: Field, message?: string) {
+    return rangeCheckN(n, x, message);
+  },
+
+  /**
+   * Checks whether the input value is in the range [0, 2^n). `n` must be a multiple of 16.
+   *
+   * This function proves that the provided field element can be represented with `n` bits.
+   * If the field element exceeds `n` bits, `Bool(false)` is returned and `Bool(true)` otherwise.
```

Comment:
this is misleading, because it only proves that the result is in range if the return value of `isInRangeN()` is true. It doesn't prove that the result is _not_ in range if the return value is false.

See my comment on where this is used: https://github.com/o1-labs/o1js/pull/1259/files#r1409932102

So maybe better to rephrase the doccomment, or frame it as a helper function for now. I can follow up with some code that actually proves the full thing, which I think would also make the UInt64 comparison cheaper (currently the comparison calls rangeCheckHelper twice, but it can be done with one call)